src/components/tables/EntityTable.tsx

```diff
@@ -10,51 +12,208 @@ import {
 import { FormattedMessage, useIntl } from 'react-intl';
-import {
-    DeploymentStatus,
-    Entity,
-    EntityMetadata,
-} from 'stores/ChangeSetStore';
+import { TABLES } from 'services/supabase';
+import { PostgrestError, useClient } from 'supabase-swr';
+
+enum TableStatuses {
+    LOADING = 'LOADING',
+    DATA_FETCHED = 'DATA_FETCHED',
+    NO_EXISTING_DATA = 'NO_EXISTING_DATA',
+    TECHNICAL_DIFFICULTIES = 'TECHNICAL_DIFFICULTIES',
+    UNMATCHED_FILTER = 'UNMATCHED_FILTER',
+}
+
+type TableStatus =
+    | TableStatuses.LOADING
+    | TableStatuses.DATA_FETCHED
+    | TableStatuses.NO_EXISTING_DATA
+    | TableStatuses.TECHNICAL_DIFFICULTIES
+    | TableStatuses.UNMATCHED_FILTER;
+
+type DeploymentStatus = 'ACTIVE' | 'INACTIVE';
 
 type SortDirection = 'asc' | 'desc';
 
 interface Props {
-    entities: Entity[];
-    updateDeploymentStatus: (
-        key: string,
-        deploymentStatus: DeploymentStatus
-    ) => void;
+    noExistingDataContentIds: {
+        header: string;
+        message: string;
+        docLink: string;
+        docPath: string;
+    };
+}
+
+interface Entity {
+    deploymentStatus: DeploymentStatus;
+    name: string;
+    catalogNamespace: string;
+    connectorType: string;
+    dateCreated: string;
+}
+
+interface TableState {
+    status: TableStatus;
+    error?: PostgrestError;
 }
 
 interface TableColumn {
-    field: keyof EntityMetadata | null;
+    field: keyof Entity | null;
     headerIntlKey: string;
 }
 
-function EntityTable({ entities, updateDeploymentStatus }: Props) {
+interface ConnectorInfo {
+    detail: string;
+    image_name: string;
+}
+
+const CONNECTORS_QUERY = `detail, image_name`;
+
+const LIVE_SPECS_QUERY = `spec_type, catalog_name, updated_at, connector_image_name, id`;
+
+function EntityTable({ noExistingDataContentIds }: Props) {
+    const supabaseClient: SupabaseClient = useClient();
+    const intl = useIntl();
+
+    const [tableState, setTableState] = useState<TableState>({
+        status: TableStatuses.LOADING,
+    });
+
+    const [connectors, setConnectors] = useState<any[] | undefined | null>(
+        null
+    );
+    const [publications, setPublications] = useState<any[] | undefined | null>(
+        null
+    );
+
+    const [unfilteredEntities, setUnfilteredEntities] = useState<
+        Entity[] | null
+    >(null);
+    const [filteredEntities, setFilteredEntities] = useState<Entity[] | null>(
+        null
+    );
+    const [entities, setEntities] = useState<Entity[] | null>(null);
+
     const [sortDirection, setSortDirection] = useState<SortDirection>('desc');
     const [columnToSort, setColumnToSort] =
-        useState<keyof EntityMetadata>('dateCreated');
+        useState<keyof Entity>('dateCreated');
 
     const [page, setPage] = useState(0);
     const rowsPerPage = 10;
     const rowHeight = 57;
 
-    const entityDetails: EntityMetadata[] = entities.map(
-        (entity) => entity.metadata
-    );
+    useEffect(() => {
```

Comment:
Upon mounting the component, the list of connectors and capture publications are fetched.